app/models/host.rb

```diff
@@ -136,6 +136,7 @@ class Host < ApplicationRecord
   virtual_column :vmm_vendor_display,           :type => :string
   virtual_column :ipmi_enabled,                 :type => :boolean
   virtual_column :archived, :type => :boolean
+  virtual_column :has_policies, :type => :boolean
```

Comment:
Can this virtual column just live directly in the miq_policy_mixin?  That way the virtual_column definition lives with the actual method that implements it.

Reply:
Yes, should be able to add the virtual column once in the mixin because it's generic
> @gtanzillo Do you know what value would work best?

It should be `:uses => {:taggings => :tags}` but that doesn't seem to help. I think it's because it's going through the class method `tag_list` which calls `Tag.list`. That method eventually calls `object.tags` and that's a `has_many` through a polymorphic association. @Fryguy

Reply:
@kbrock Would you be able to help us with this? I'm wondering whether there is a more performant approach to determine whether an object has policies assigned, or perhaps there isn't. But was thinking we can see if it can be done with pure SQL with joins and `LIKE` clauses, the assignment code is really difficult to deal with.

Also, @h-kataria was saying that she is not able to sort on that virtual column and it messes up the list view after it attempts to do so. We think it's due to this logic [here](https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_report/search.rb#L19-L20). Perhaps solving the virtual column with SQL would help with sorting?

Reply:
NOTE: I think I wrote that comment. And I think it should read virtual_relation not virtual_column/virtual_attribute.

We like to put columns into the query. But if it is only accessible though virtual_relation (aka through ruby) we are not able to put it into the query.

Yes, creating a virtual_attribute will avoid this whole mess.
Also sorting by a virtual attribute will allow us to not have to download all data and sort in ruby - a big win.